packages/fx-core/src/component/generator/utils.ts

```diff
@@ -118,9 +111,9 @@ export async function fetchTemplateZipUrl(
   tryLimits = defaultTryLimits,
   timeoutInMs = defaultTimeoutInMs
 ): Promise<string> {
-  const selectedTag = await selectTemplateTag(async () =>
-    (await fetchTagList(templateTagListURL, tryLimits, timeoutInMs)).replace(/\r/g, "").split("\n")
-  );
+  const templateTagListURL = templateConfig.tagListURL;
+  const tags = await fetchTagList(templateTagListURL, tryLimits, timeoutInMs);
```

Comment:
I suppose making `fetchTagList` as a callback function can improve performance.
When it is alpha version, generator does not need to fetch tag list and use fallback template immediately.
While it indeed hurts the code readability. You can make the decision.

Reply:
You are absolutely right. I will use a callback function here.